src/hera.cpp

```diff
@@ -229,6 +230,20 @@ void hera_destroy_result(evmc_result const* result) noexcept
   delete[] result->output_data;
 }
 
+WasmEngine hera_instantiate_engine(hera_wasm_engine wasm_engine)
```

Comment:
@chfast this was my idea for abstracting it, but this requires a non-virtual constructor in `WasmEngine`. Looking forward your advice.

Reply:
You cannot do it this way. It will copy only the WasmEngine part of the create object (this is called object slicing, theoretically not always an error, but here yes).

You have to return `std::unique_ptr<WasmEngine>`.